src/pages/presentations/building-decentralised-applications.md

```diff
@@ -0,0 +1,862 @@
+
+---
+
+## Ethers Contract Object
+
+```javascript
+import ethers from "ethers";
+import erc20Abi from "./erc20-abi.json";
+
+const RPC = "https://my-cool-rpc.com/";
+const USER = "0x15463F7566d797a4b36517eB3A1cAFaB58f1A381";
+const CHASE = "0x3A61da6D37493E2f248A6832F49b52Af0a6f4Fbb";
+const UNI_ADDRESS = "0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984";
+const PRIVATE_KEY = "[[users private key]]";
+
+const provider = new ethers.JsonRpcProvider(RPC);
+const signer = new ethers.Wallet(pk, PROVIDER);
+
+const uniContract = new ethers.Contract(UNI_ADDRESS, erc20Abi, signer);
+
+const uniBalance = await uniContract.balanceOf(USER);
+
+const tx = await uniContract.transfer(CHASE, uniBalance);
+
+console.log(tx.hash); // Logs: 0xbed1ab34efcea49dd55c51134916a6ea8b28967fe32509573039e565102f2a75
+
+const receipt = await tx.wait();
+
+console.log(receipt); // Logs: (an object with transaction data)
+```
+
+???
+
+We'll now introduce another feature of Ethers which is commonly used which is the Contract object. This allows you to create a TypeScript object that represents your smart contract on chain. Which you can then use to conveniently query views from your contract. Or call functions to create transactions.
+
+We have a new import at the start here, it is importing the ABI for an ERC20 contract. As a reminder, an ABI is a JSON object that lists all of the functions, views and events that a smart contract has. We have a couple of new constants here, one is a dummy address for myself, one is the Uniswap Token UNI and one is the user's private key. Which we would never store in plain text like this, but this is just for an example. Just further down we have the creation of the Contract object for the UNI Token. We pass as an input here the address of the token, the ABI, and the signer. A signer is similar to a provider, other than it also allows for posting transactions for the user. We could also create this contract passing through the provider here instead, and the contract object would still work, but would only allow for reading views.
+
+From there it's just one line to query a view, so here we're getting the user's balance of the UNI token. And it's also just one line to call a function and raise a transaction, here we're doing that to call the transfer function to send the users UNI balance to myself. Optionally, calling a function returns a transaction object, which we can then wait to load until the transaction has been executed on chain.
+
+So you can see with this code, you could now use this to link up to a front end to display some data about the user's state on the blockchain. And with the press of a button you could transfer tokens to another address.
+
+---
+
+## wagmi
+
+- wagmi is yet another abstraction layer for blockchain integration
+- It focuses mostly on integrating with React front ends
```

Comment:
Thanks have added a small change to explain what React is in the slide. And will also try use the term 'function' instead of 'hook' when speaking as it's kind of the same thing but less jargon.